vxfreeswitch/tests/helpers.py

```diff
@@ -0,0 +1,325 @@
+        if "name" in d:
+            params["execute-app-name"] = d.get("name")
+        if "arg" in d:
+            params["execute-app-arg"] = d.get("arg")
+        return cls(cmd_type, params)
+
+
+class EslParser(object):
+    """
+    Simple in-efficient parser for the FreeSwitch eventsocket protocol.
+    """
+
+    def __init__(self):
+        self.data = ""
+
+    def parse(self, new_data):
+        data = self.data + new_data
+        cmds = []
+        while "\n\n" in data:
+            cmd_data, data = data.split("\n\n", 1)
+            command = EslCommand("unknown")
+            first_line = True
+            for line in cmd_data.splitlines():
+                line = line.strip()
+                if not line:
+                    continue
+                if first_line:
+                    command.cmd_type = line.strip()
+                    first_line = False
+                    continue
+                if ":" in line:
+                    key, value = line.split(":", 1)
+                    command[key] = value.strip()
+            cmds.append(command)
+        self.data = data
+        return cmds
+
+
+class EslTransport(StringTransport):
+
+    def __init__(self):
+        StringTransport.__init__(self)
+        self.cmds = DeferredQueue()
+        self.esl_parser = EslParser()
+
+    def write(self, data):
+        StringTransport.write(self, data)
+        for cmd in self.esl_parser.parse(data):
+            self.cmds.put(cmd)
+
+
+class RecordingServer(Protocol):
+    def __init__(self):
+        self.command_parser = EslParser()
+
+    def connectionMade(self):
+        self.factory.clients.append(self)
+
+    def _send_event(self, content):
+        self.transport.write(
+            'Content-Length: %s\n' % len(content) +
+            'Content-Type: text/event-plain\n\n' +
+            content)
+
+    def dataReceived(self, line):
+        commands = self.command_parser.parse(line)
+        #self.factory.data.extend(commands)
+        for cmd in commands:
+            response = self.factory.get_response(cmd)
+            self.transport.write(response.to_bytes())
+            continue
+            uuid = '%s' % self.factory.uuid()
+            # Send job received correctly
+            self.transport.write(
+                'Content-Type: command/reply\n'
+                'Reply-Text: +OK\n'
+                'Job-UUID: %s\n\n' % uuid)
+            if cmd.cmd_type.startswith('bgapi originate'):
+                # Send job complete success response
+                if self.factory.fail_connect:
+                    content_body = '+ERROR %s\n' % uuid
+                else:
+                    content_body = '+OK %s\n' % uuid
+                content = (
+                    'Content-Length: %d\n'
+                    'Event-Name: BACKGROUND_JOB\n'
+                    'Job-UUID: %s\n\n'
+                    '%s' %
+                    (len(content_body), uuid, content_body))
+                self._send_event(content)
```

Comment:
This whole block should be removed, since it is skipped using `continue`, and since commands are now handled by the factory.